examples/Makefile.testing

```diff
@@ -0,0 +1,178 @@
+	@$(ECHO) -e "Modifying object-detection to use new object-images repo and adding images...";
+	$(PACHCTL) create repo object-images
+	$(JQ) '$(STATS)|$(DETECTMODEL)|$(DETECTIMAGES)' ml/object-detection/detect.json  | $(PACHCTL) update pipeline -f - >/dev/null
+	$(PACHCTL) put file images@master -i opencv/images4.txt
+	$(PACHCTL) put file object-images@master:dogs.jpg -f ml/object-detection/images/dogs.jpg
+	$(PACHCTL) put file object-images@master:airplane.jpg -f ml/object-detection/images/airplane.jpg
+	../etc/examples/wait-for-jobs.sh succeeded-only "Done adding new images in object-images@master and images@master."
+	@$(ECHO) -e "Adding data to gatk...";
+	$(PACHCTL) start transaction > /dev/null
+	$(PACHCTL) start commit reference@master > /dev/null 2>/dev/null
+	$(PACHCTL) start commit samples@master >/dev/null 2>/dev/null
+	$(PACHCTL) finish transaction > /dev/null
+	$(PACHCTL) put file reference@master:ref.dict  -f gatk/GATK_Germline/data/ref/ref.dict
+	$(PACHCTL) put file reference@master:ref.fasta -f gatk/GATK_Germline/data/ref/ref.fasta
+	$(PACHCTL) put file reference@master:ref.fasta.fai -f gatk/GATK_Germline/data/ref/ref.fasta.fai
+	$(PACHCTL) put file reference@master:refSDF -r -f gatk/GATK_Germline/data/ref/refSDF
+	$(PACHCTL) put file samples@master:mother/mother.bam -f gatk/GATK_Germline/data/bams/mother.bam
+	$(PACHCTL) put file samples@master:mother/mother.bai -f gatk/GATK_Germline/data/bams/mother.bai
+	$(PACHCTL) finish commit reference@master > /dev/null
+	$(PACHCTL) finish commit samples@master > /dev/null
+	../etc/examples/wait-for-jobs.sh succeeded-only "Done adding initial gatk data in references and master."
+	if GATK_FATHER_COMMIT=`$(PACHCTL) start commit samples@master` && \
+	   [ -n $${GATK_FATHER_COMMIT} ] ; \
+	then \
+		$(ECHO) -e "Adding father to gatk, samples@$${GATK_FATHER_COMMIT}, we will delete his data later..." ; \
+		$(PACHCTL) put file samples@master:father/father.bam -f gatk/GATK_Germline/data/bams/father.bam ; \
+		$(PACHCTL) put file samples@master:father/father.bai -f gatk/GATK_Germline/data/bams/father.bai ; \
+		$(PACHCTL) finish commit samples@master > /dev/null ; \
+		GATK_SON_COMMIT=`$(PACHCTL) start commit samples@master` ; \
+		$(ECHO) -e "Adding son, samples@$${GATK_SON_COMMIT}..." ; \
+		$(PACHCTL) put file samples@master:son/son.bam -f gatk/GATK_Germline/data/bams/son.bam ; \
+		$(PACHCTL) put file samples@master:son/son.bai -f gatk/GATK_Germline/data/bams/son.bai	; \
+		$(PACHCTL) finish commit samples@master > /dev/null ; \
+		$(ECHO) -e "Deleting father commit, samples@$${GATK_FATHER_COMMIT}..." ; \
+		$(PACHCTL) delete commit samples@$${GATK_FATHER_COMMIT} ; \
+	else \
+		$(ECHO) -e "Error: could not create commit in samples@master." ; \
+		exit 1 ; \
+	fi
+	../etc/examples/wait-for-jobs.sh succeeded-only "All jobs finished. Inspect output above to determine if cluster ready for testing."
```

Comment:
I think once there's control structures, the makefile target should be broken out into a separate script

Reply:
That's an interesting approach.  I'm not sure I agree with it fully.  There are circumstances, like here and on lines 102-116, where a simple control structure in the makefile makes it easy to fully understand the flow of the test.   The creation of the father and son commits, and then the deletion of the father commit, are part of this  cluster setup, not a function that can be called again and again.

So, in this case (and at the beginning of the target on lines 102-116), breaking it out makes the target _less_ understandable.  The flow is an integral part of this particular test, so the logic belongs in the test.